modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm

```diff
@@ -33,212 +51,433 @@
+ */
+/datum/action/innate/alter_form/proc/alter_colours(mob/living/carbon/human/alterer)
+	var/color_choice = show_radial_menu(
+		alterer,
+		alterer,
+		list(
+			"Primary" = image(icon = 'modular_skyrat/master_files/icons/mob/actions/actions_slime.dmi', icon_state = "slime_red"),
+			"Secondary" = image(icon = 'modular_skyrat/master_files/icons/mob/actions/actions_slime.dmi', icon_state = "slime_green"),
+			"Tertiary" = image(icon = 'modular_skyrat/master_files/icons/mob/actions/actions_slime.dmi', icon_state = "slime_blue"),
+			"All" = image(icon = 'modular_skyrat/master_files/icons/mob/actions/actions_slime.dmi', icon_state = "slime_rainbow"),
+		),
+		tooltips = TRUE,
+	)
+	if(!color_choice)
+		return
+	var/color_target
+	switch(color_choice)
+		if("Primary", "All")
+			color_target = "mcolor"
+		if("Secondary")
+			color_target = "mcolor2"
+		if("Tertiary")
+			color_target = "mcolor3"
+
+	var/new_mutant_colour = input(
+		alterer,
+		"Choose your character's new [color_choice = "All" ? "" : lowertext(color_choice)] color:",
+		"Form Alteration",
+		alterer.dna.features[color_target]
+	) as color|null
+	if(!new_mutant_colour)
+		return
+
+	var/marking_reset = tgui_alert(
+		alterer,
+		"Would you like to reset your markings to match your new colors?",
+		"Reset markings",
+		list("Yes", "No"),
+	)
+	var/mutant_part_reset = tgui_alert(
+		alterer,
+		"Would you like to reset your mutant body parts(not limbs) to match your new colors?",
+		"Reset mutant parts",
+		list("Yes", "No"),
+	)
+	var/hair_reset = tgui_alert(
+		alterer,
+		"Would you like to reset your hair to match your new colors?",
+		"Reset hair",
+		list("Hair", "Facial Hair", "Both", "None"),
+	)
+
+	if(color_choice == "All")
+		alterer.dna.features["mcolor"] = sanitize_hexcolor(new_mutant_colour)
+		alterer.dna.features["mcolor2"] = sanitize_hexcolor(new_mutant_colour)
+		alterer.dna.features["mcolor3"] = sanitize_hexcolor(new_mutant_colour)
+		alterer.dna.update_uf_block(DNA_MUTANT_COLOR_BLOCK)
+		alterer.dna.update_uf_block(DNA_MUTANT_COLOR_2_BLOCK)
+		alterer.dna.update_uf_block(DNA_MUTANT_COLOR_3_BLOCK)
+	else
+		alterer.dna.features[color_target] = sanitize_hexcolor(new_mutant_colour)
+		switch(color_target)
+			if("mcolor")
+				alterer.dna.update_uf_block(DNA_MUTANT_COLOR_BLOCK)
+			if("mcolor2")
+				alterer.dna.update_uf_block(DNA_MUTANT_COLOR_2_BLOCK)
+			if("mcolor3")
+				alterer.dna.update_uf_block(DNA_MUTANT_COLOR_3_BLOCK)
+
+	if(marking_reset && marking_reset == "Yes")
+		for(var/zone in alterer.dna.species.body_markings)
+			for(var/key in alterer.dna.species.body_markings[zone])
+				var/datum/body_marking/iterated_marking = GLOB.body_markings[key]
+				if(iterated_marking.always_color_customizable)
+					continue
+				alterer.dna.species.body_markings[zone][key] = iterated_marking.get_default_color(alterer.dna.features, alterer.dna.species)
+
+	if(mutant_part_reset && mutant_part_reset == "Yes")
```

Comment:
same thing as above: what is the need to check for something if you are checking it for "yes" directly afterward.